From dd085d0ea5342c98d98c473ffe5323bd04c60252 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 8 Mar 2023 15:32:11 +0100
Subject: [PATCH 285/841] feat(stm32mp2-fdts): add firewall nodes in fw-config

The stm32mp257f-ev board fw-config DT file now includes a dtsi file that
describes the DDR firewall configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 2227377b54242391dbadbbec6c4abb0a96423c56 feat(stm32mp2-fdts): add firewall nodes in fw-config]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I3300b00ba6e59b8df4f02db531edd590683c333f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/294405
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 fdts/stm32mp25-fw-config.dtsi                |  6 +++++
 fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi | 23 ++++++++++++++++++++
 2 files changed, 29 insertions(+)
 create mode 100644 fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi

diff --git a/fdts/stm32mp25-fw-config.dtsi b/fdts/stm32mp25-fw-config.dtsi
index 218eb2b80..d023378a0 100644
--- a/fdts/stm32mp25-fw-config.dtsi
+++ b/fdts/stm32mp25-fw-config.dtsi
@@ -53,4 +53,10 @@
 			id = <BL32_IMAGE_ID>;
 		};
 	};
+
+	st_mem_firewall: st-mem-firewall {
+		compatible = "st,stm32mp2-mem-firewall";
+		#address-cells = <2>;
+		#size-cells = <2>;
+	};
 };
diff --git a/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi b/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi
new file mode 100644
index 000000000..d9d928208
--- /dev/null
+++ b/fdts/stm32mp257f-ev-ca35tdcid-fw-config.dtsi
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
+ */
+
+/*
+ * STM32MP25 tf-a firmware config
+ * Project : eco
+ * Generated by XLmx tool version 2.2 - 3/8/2023 2:34:26 PM
+ */
+
+&st_mem_firewall {
+	bl31_context: bl31-context@81ff0000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00000000 0x81ff0000 0x00000000 0x10000>;
+		st,protreg = <RISAFPROT(RISAF_REG_ID(7), RIF_CID1_BF, RIF_CID1_BF, RIF_CID1_BF, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+	};
+	op_tee: op-tee@82000000 {
+		compatible = "shared-dma-pool";
+		reg = <0x00000000 0x82000000 0x00000000 0x2000000>;
+		st,protreg = <RISAFPROT(RISAF_REG_ID(8), RIF_CID1_BF, RIF_CID1_BF, 0, RIF_SEC, RIF_ENC_EN, RIF_BREN_EN)>;
+	};
+};
-- 
2.47.2

